File: pancakeSort.cpp
class Solution {
public:
    vector<int> pancakeSort(vector<int>& A) {
        vector<int> flips;
        for(int i=0; i<A.size(); i++){
            auto max_pos = max_element(A.begin(),A.end()-i);
            flips.push_back(max_pos-A.begin()+1);
            flips.push_back(A.size()-i);
            reverse(A.begin(), max_pos+1);
            reverse(A.begin(), A.end()-i);
        }
        return flips;
    }
};

File: pancakeSort.py
class Solution:
    def pancakeSort(self, A: List[int]) -> List[int]:
        if len(A)==1:
            return []
        max_pos = A.index(max(A))
        A[0:max_pos+1] = A[0:max_pos+1][::-1]
        A = A[::-1]
        return [max_pos+1, len(A)] + self.pancakeSort(A[:-1])
